### PR TITLE
influxdb: 1.8.10 -> 1.10.0

### PR DESCRIPTION
**Pull request by knl**
###### Description of changes

This change updates the version of influxdb and also makes it compile on darwin. The issue with darwin was that 1.8 couldn't be compiled using go 1.18. 1.10 can be compiled using the recent go.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/nosql/influxdb/default.nix [hunk 1]
```diff
@@ -1,17 +1,81 @@
...
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-PErAxRpSi1Kk6IpEAhsUSxCGYeY4p6bbhwLdbRB0M00=";
+    sha256 = "sha256-BMHR9EdYC+8oA0he7emzBRmNnHn15nO/5NqsLcr+R0k=";
   };
 
-  vendorSha256 = "sha256-jgAbEWXL1LYRN7ud9ij0Z1KBGHPZ0sRq78tsK92ob8k=";
+  vendorSha256 = "sha256-AY04cmfg7vbrWR4+LBuCFYqBgQJBXlPpO+2oj0qqjM4=";
+
+  nativeBuildInputs = [ bash pkg-config ];
```
Comment (zowoq) [inline pkgs/servers/nosql/influxdb/default.nix:62]: Not sure why bash is needed here? Might want to use `patchShebangs`?

File: pkgs/servers/nosql/influxdb/default.nix [hunk 2]
```diff
@@ -1,17 +1,81 @@
...
+
+  PKG_CONFIG_PATH = "${flux}/pkgconfig";
+
+  prePatch = ''
+    substituteInPlace Makefile \
+      --replace './pkg-config.sh' $src/pkg-config.sh
```
Comment (zowoq) [inline pkgs/servers/nosql/influxdb/default.nix:68]: `'./pkg-config.sh' $src/pkg-config.sh` Aren't these the same file?

File: pkgs/servers/nosql/influxdb/default.nix (hunk 1 above)
Comment (knl) [reply pkgs/servers/nosql/influxdb/default.nix:62]: True, I can get rid of it.

File: pkgs/servers/nosql/influxdb/default.nix (hunk 2 above)
Comment (knl) [reply pkgs/servers/nosql/influxdb/default.nix:68]: They are, but the former causes an error, while with the full path it works.

Comment (zowoq) [reply pkgs/servers/nosql/influxdb/default.nix:68]: Just noticed that make and the makefile isn't even used during the build.

Comment (knl) [reply pkgs/servers/nosql/influxdb/default.nix]: You're absolutely right, I've removed it now.

**Pull request merged**